assignments/array-methods.js

```diff
@@ -2,82 +2,512 @@
+		email: 'balway1c@ow.ly',
+		shirt_size: 'S',
+		company_name: 'Voolia',
+		donation: 107
+	},
+	{
+		id: 50,
+		first_name: 'Shell',
+		last_name: 'Baine',
+		email: 'sbaine1d@intel.com',
+		shirt_size: 'M',
+		company_name: 'Gabtype',
+		donation: 171
+	}
+];
 
 // ==== Challenge 1: Use .forEach() ====
-// The event director needs both the first and last names of each runner for their running bibs.  Combine both the first and last names into a new array called fullName. 
-let fullName = [];
-console.log(fullName);
+// The event director needs both the first and last names of each runner for their running bibs.  Combine both the first and last names into a new array called fullName.
+const fullName = [];
+
+// Cycle over runners array and get first and last name.
+runners.forEach(runner => {
+	fullName.push([`${runner.first_name} ${runner.last_name}`]);
+});
+
+console.table(fullName);
 
 // ==== Challenge 2: Use .map() ====
 // The event director needs to have all the runner's first names converted to uppercase because the director BECAME DRUNK WITH POWER. Convert each first name into all caps and log the result
-let allCaps = [];
-console.log(allCaps); 
+const allCaps = runners.map(runner => runner.first_name.toUpperCase());
+console.table(allCaps);
 
 // ==== Challenge 3: Use .filter() ====
 // The large shirts won't be available for the event due to an ordering issue.  Get a list of runners with large sized shirts so they can choose a different size. Return an array named largeShirts that contains information about the runners that have a shirt size of L and log the result
-let largeShirts = [];
-console.log(largeShirts);
+const largeShirts = runners.filter(runner => runner.shirt_size === 'L');
+console.table(largeShirts);
 
 // ==== Challenge 4: Use .reduce() ====
 // The donations need to be tallied up and reported for tax purposes. Add up all the donations into a ticketPriceTotal array and log the result
-let ticketPriceTotal = [];
+
+// Note: 0 is the starting point for the accumulator
+const ticketPriceTotal = runners.reduce((accumulator, current) => accumulator + current.donation, 0);
 console.log(ticketPriceTotal);
 
 // ==== Challenge 5: Be Creative ====
 // Now that you have used .forEach(), .map(), .filter(), and .reduce().  I want you to think of potential problems you could solve given the data set and the 5k fun run theme.  Try to create and then solve 3 unique problems using one or many of the array methods listed above.
 
 // Problem 1
+const donorAboveTwoHeundred = runners.filter(runner => runner.donation >= 200);
+console.table(donorAboveTwoHeundred);
 
 // Problem 2
+const shirtSizes = runners.map(curr => curr.shirt_size);
+const allSizes = Array.from(new Set(shirtSizes));
+console.log('------------All unique sizes---------');
+console.table(allSizes);
+
+// Problem 3
+let message = '';
+runners.forEach(curr => {
+	message += `----------------------
+  Donation Appreciation
+  -----------------------
```

Comment:
Great job with this!

<img width="443" alt="Capture d’écran 2019-05-08 à 11 42 07" src="https://user-images.githubusercontent.com/2175908/57365980-6248b800-7186-11e9-9d58-fd2aae9698e5.png">